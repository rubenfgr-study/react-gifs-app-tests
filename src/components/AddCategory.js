import pt from "prop-types";
import { useState } from "react";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);

    console.log("handleInputchange launch");
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <h3>{inputValue}</h3>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: pt.func.isRequired,
};
