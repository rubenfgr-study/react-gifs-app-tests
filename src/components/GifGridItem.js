import "animate.css";
import propTypes from "prop-types";
import React from "react";

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeInRight">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifGridItem.prototypes = {
  title: propTypes.string.isRequired,
  url: propTypes.string.isRequired,
};
