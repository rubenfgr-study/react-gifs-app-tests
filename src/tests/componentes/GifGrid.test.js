import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("GifGrid Test", () => {
  const category = "category";

  test("GifGrid test snapshoot", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar items cuando se cargan imÃ¡genes useFetchGifs", () => {
    const gifs = [
      { id: "abc", url: "https://localhost/img.jpg", title: "ABC" },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBeFalsy();
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
