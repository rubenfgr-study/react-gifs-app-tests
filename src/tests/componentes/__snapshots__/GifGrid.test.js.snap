// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Test GifGrid test snapshoot 1`] = `
<Fragment>
  <h3>
    category
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid Test debe mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3>
    category
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="ABC"
      url="https://localhost/img.jpg"
    />
  </div>
</Fragment>
`;
